src/utils/media-utils.js

```diff
@@ -142,10 +143,16 @@ export function getTrackDurationFromSourceDuration(sourceDuration, playbackRange
   return (playbackRange[1] - playbackRange[0]) * sourceDuration;
 }
 
-export function getFullSourceUrl({ url, sourceType, cdnRootUrl }) {
-  return url && sourceType === MEDIA_SOURCE_TYPE.internal
-    ? `${cdnRootUrl}/${url}`
-    : url || null;
+export function getMediaSourceType({ sourceUrl, cdnRootUrl }) {
+  if (sourceUrl.startsWith(cdnRootUrl) || sourceUrl.startsWith(CDN_URL_PREFIX)) {
+    return MEDIA_SOURCE_TYPE.internal;
+  }
+
+  if (sourceUrl.startsWith('https://www.youtube.com/')) {
+    return MEDIA_SOURCE_TYPE.youtube;
+  }
+
+  return MEDIA_SOURCE_TYPE.external;
```

Comment:
Nice! One step closer to dropping the `sourceType`.

Reply:
:+1: yeah